core/src/main/scala/pureconfig/BasicReaders.scala

```diff
@@ -44,6 +46,15 @@ trait UriAndPathReaders {
 }
 
 /**
+ * Trait containing `ConfigReader` instances for classes related to regular expressions.
+ */
+trait RegexReaders {
+
+  implicit val patternReader = ConfigReader.fromNonEmptyString[Pattern](catchReadError(Pattern.compile))
```

Comment:
Shouldn't we allow empty strings to be parsed as regexes? It may not be an overly useful regex, but it's still a valid one. Also, as `patternWriter` can write an empty string, a round-trip test would require `patternReader` to read them, too.